stft

abhayKashyap03 opened the pull request:
Close #8827

YushaArif99 commented on ivy/container/experimental/layers.py:
```diff
@@ -942,3 +942,67 @@ def dct(
             norm=norm,
             out=out,
         )
+
```
Could you also write the docstring example for both container instance and static methods

YushaArif99 commented on ivy/functional/ivy/experimental/layers.py:
```diff
@@ -619,3 +619,22 @@ def dropout1d(
     return ivy.current_backend(x).dropout1d(
         x, prob, training=training, data_format=data_format, out=out
     )
+
```
could you also add docstring examples for the ivy function as well

YushaArif99 reviewed (changes requested):
Hi @abhayKashyap03 

Sorry for the late reply. Was a little swamped with work and so reviewing this PR totally slipped under my radar. 
Overall, the changes look good to me. Could you kindly add the docstring examples as well in the places I have pointed out above. Also, I dont see an ivy.Array implementation of the `stft()` function anywhere. If you've added a container implementation, you should also add the implementation of this function for the `ivy.Array` class as well. Could you kindly add that as well. Thanks! 🙂 
Edit: Could you also fix the lint formatting errors? Thanks!

abhayKashyap03 commented:
> Hi @abhayKashyap03
> 
> Sorry for the late reply. Was a little swamped with work and so reviewing this PR totally slipped under my radar. Overall, the changes look good to me. Could you kindly add the docstring examples as well in the places I have pointed out above. Also, I dont see an ivy.Array implementation of the `stft()` function anywhere. If you've added a container implementation, you should also add the implementation of this function for the `ivy.Array` class as well. Could you kindly add that as well. Thanks! 🙂 Edit: Could you also fix the lint formatting errors? Thanks!

Hi @YushaArif99, 
You asked me to add the implementation for `ivy.Array` class too, I'm assuming you meant the `array` folder under `ivy`. I've added that along with the docstrings in the latest commit, please review and let me know if I need to make any more changes.

YushaArif99 reviewed (changes requested):
Hi @abhayKashyap03 
Thanks, a lot for making the changes. Could you kindly revert the changes made to the `.idea/` files? Other than that everything seems good and hopefully the PR will be ready to be merged (once all CI tests are passed). Thanks!

YushaArif99 commented on ivy/container/experimental/layers.py:
```diff
@@ -942,3 +942,143 @@ def dct(
+        ret
+            Container with transformed signals
+
+        """
```
One small addition. Could you kindly also add a docstring example for `static_stft`. You may reuse the example you used in the `stft` function below. The only change would be that instead of calling `x.stft(...)`, you'd now call `ivy.Container.static_stft(...)`.

YushaArif99 reviewed (changes requested):
Hi @abhayKashyap03 

The merge conflicts you're getting are probably because your version of the `/layers.py` files is outdated as some new commits might have been made to those files on the `master` branch.
A quick fix that I use is to simply copy-paste the content of those files from `master` to the corresponding files in this PR  and then append the new changes you added to the bottom of those files. Just be sure to copy the changes you made to Notepad etc. before you overwrite these files.  

Let me know if you have any other questions 😃

abhayKashyap03 commented:
@YushaArif99 I just downloaded the .idea folder from the original ivy repo and replaced my forked folder with the original, so I'm hoping that is enough to revert all the changes it says I've made. 

I also still see the conflicts with the layers.py files, even though I copy-pasted the contents of the original files and added my changes, could you please look into this and let me know why this is happening? I also noticed that when I go to my forked ivy repo, it says my branch is 6 commits ahead and >500 commits behind the master branch of the ivy repo, but when I execute `git pull`, it says "already up to date".  Let me know what I can do to resolve these conflicts.

YushaArif99 reviewed (changes requested):
Hi @abhayKashyap03 

I can seee that some of your recent commits have now modified even more files than before and there are now a ton of merge conflicts 😅 

I think in this case, here's what you try. 
- First keep a backup of all the changes you made (so that you can later add them again afterwords)
-  Now, use the `git reset <commit hash>` (first read about how to use it) command to reset your branch to now point to a specific commit in the git history. Basically, you have made 6 commits it seems. I think if you reset to the 1st commit you made, at that point in time, you had not made any changes to the `.idea` folder and so these new unwanted changes should get reverted.
- Now, you can add back the new changes you wanted to add
- `git add`  the changes followed by `git commit` and `git push`

If all this doesnt work, then yeah you do have the option of creating a new fork and adding the changes there. Do note however, that it would require creating a new PR (since you'd now have a new branch basically) and so you should close this one.

abhayKashyap03 commented:
@YushaArif99 I am opening a new PR instead of this, I think at this point, there have been too many commits and changes made to too many files, including all the work others candidates and engineers have been doing. I am going to create a new fork with the latest version of the original repo and open a new PR with that fork. I'll mention the new PR here so that it's easier for you to find it.
